### PR TITLE
Power Fx connector improvements

### PR DESCRIPTION
**Pull request by LucGenetier**
Connectors: 
- allow use of Swagger document in lieu of endpoint (retrieve it automatically, when available)
- disallow endpoints starting with http://

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/tests/Microsoft.PowerFx.Connectors.Tests/PowerPlatformConnectorTests.cs [hunk 1]
```diff
@@ -16,23 +17,36 @@ namespace Microsoft.PowerFx.Tests
...
+                    apiDoc,                                                     // Swagger file
+                    "839eace6-59ab-4243-97ec-a5b8fcc104e4",                     // environment
+                    "shared-msnweather-8d08e763-937a-45bf-a2ea-c5ed-ecc70ca4",  // connectionId
+                    () => "AuthToken1",
+                    httpClient)
+                {
+                    SessionId = "MySessionId"
+                }
+                : new PowerPlatformConnectorClient(
+                    "firstrelease-001.azure-apim.net",                          // endpoint
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Connectors.Tests/PowerPlatformConnectorTests.cs:42]: We're not allowing https:// for the endpoint. 
Can we add a test to ensure we get the header below correct even when endpoint has https:// ?  (the header doesn't have https)

authority: firstrelease-001.azure-apim.net

File: src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs [hunk 2]
```diff
@@ -16,8 +16,12 @@ namespace Microsoft.PowerFx.Connectors
     // https://docs.microsoft.com/en-us/connectors/custom-connectors/openapi-extensions#x-ms-visibility
     public static class OpenApiExtensions
     {
-        public static string GetBasePath(this OpenApiDocument openApiDocument)
-        {            
+        public static string GetBasePath(this OpenApiDocument openApiDocument) => GetUriElement(openApiDocument, (uri) => uri.PathAndQuery);
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs:19]: Can we add unit tests specifically for these methods?

**Comment by BruceHaley** [comment]
:x: 2 Binary Compatibility issues for **Microsoft.PowerFx.Connectors** compared against [version 0.2.2-preview.20220803-315292](https://www.nuget.org/packages/Microsoft.PowerFx.Connectors/0.2.2-preview.20220803-315292).
<details>

```
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.PowerPlatformConnectorClient.GetAuthToken.get()' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.PowerPlatformConnectorClient.Transform(System.Net.Http.HttpRequestMessage)' does not exist in the implementation but it does exist in the contract.
```

</details>
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:x: 2 Binary Compatibility issues for **Microsoft.PowerFx.Connectors** compared against [version 0.2.2-preview.20220803-315292](https://www.nuget.org/packages/Microsoft.PowerFx.Connectors/0.2.2-preview.20220803-315292).
<details>

```
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.PowerPlatformConnectorClient.GetAuthToken.get()' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.PowerPlatformConnectorClient.Transform(System.Net.Http.HttpRequestMessage)' does not exist in the implementation but it does exist in the contract.
```

</details>
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by BruceHaley** [comment]
:x: 2 Binary Compatibility issues for **Microsoft.PowerFx.Connectors** compared against [version 0.2.2-preview.20220803-315292](https://www.nuget.org/packages/Microsoft.PowerFx.Connectors/0.2.2-preview.20220803-315292).
<details>

```
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.PowerPlatformConnectorClient.GetAuthToken.get()' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.PowerPlatformConnectorClient.Transform(System.Net.Http.HttpRequestMessage)' does not exist in the implementation but it does exist in the contract.
```

</details>
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Pull request merged**